feat: add cheesy drive

fibonacci61 opened the pull request.

Tropix126 reviewed (changes requested):
I have a few thoughts regarding this, but it mostly looks okay.

- First, this should probably go into `evian_motion`. `evian_drivetrain` isn't really intended to have any algorithm implementations for stability reasons. Driver control related things that just update the drivetrain voltages once are pretty much just non-`async` motion functions. As such, the API could be adjusted to `curvature.update(&mut self.drivetrain, throttle, turn)`, where it sets the drivetrain voltages rather than returning them.

- Second, I think that "curvature drive" is probably the more recognized name for this algorithm, at least from my experience, so it might be better to name it that.

fibonacci61 commented:
OK, I've renamed the struct to "CurvatureDrive", moved it to evian-motion::curvature. Additionally, the update function signature is now fn(&mut self, &mut Differential, ...) -> Result<(), MotorError>

Tropix126 reviewed (approved):
I've made some adjustments to match the rest of the motion APIs and am pretty happy with this. I haven't hardware tested this yet, but i'll go ahead and merge it anyways, then come back and fix it if anything ends up broken. Thanks for the contribution!
